### PR TITLE
Add automatic deploy for development environment

### PR DESCRIPTION
**Pull request by georgysavva**
This PR adds ansible deployment to our newly created devnet machines. 
The deployment is automatically triggered after the new `iotaledger/goshimmer:develop` image is built on merge into the "develop" branch.

### CODE REVIEW

File: deploy/ansible/roles/metrics/files/prometheus.yml
Comment (capossele) [inline deploy/ansible/roles/metrics/files/prometheus.yml:14]: Are we sure having the name of these machines under the `devnet` domain is not going to create confusion with the public DevNet?

File: deploy/ansible/roles/goshimmer-node/templates/docker-compose-entrynode.yml.j2
Comment (capossele) [inline deploy/ansible/roles/goshimmer-node/templates/docker-compose-entrynode.yml.j2:20]: I would add, just to be more sure that we will never ever mix this network with the public DevNet also a different networkVersion (e.g., `--autopeering.networkVersion=8000`).
And the same should also be replicated for all the other GoShimmer nodes

**Review by capossele — CHANGES_REQUESTED** [review]
Very nice work!
Just left few remarks

File: deploy/ansible/roles/metrics/files/prometheus.yml
Comment (georgysavva) [reply deploy/ansible/roles/metrics/files/prometheus.yml:14]: Yeah, I think we should settle on distinct naming for all those environments:
1. Production environment (which is deployed manually with a new release) - **DevNet**
2. Development environment (which is deployed automatically from "develop" branch) - **TestNet** ???
@karimodm @jonastheis @Wollac @luca-moser

File: deploy/ansible/roles/goshimmer-node/templates/docker-compose-entrynode.yml.j2
Comment (georgysavva) [reply deploy/ansible/roles/goshimmer-node/templates/docker-compose-entrynode.yml.j2:20]: Sure

File: .github/workflows/docker-develop.yml
Comment (Wollac) [inline .github/workflows/docker-develop.yml:9]: Why does this need to run at pull requests? We create a lot of PR with this and that wouldn't this cause issues?

File: .github/workflows/docker-develop.yml
Comment (Wollac) [inline .github/workflows/docker-develop.yml:54]: Admittedly, I don't know much about ansible and co, but have the private ssh key accessible from GHA seems like a security risk as someone could create a PR that does anything on the target machine.

File: deploy/ansible/roles/metrics/files/prometheus.yml
Comment (capossele) [reply deploy/ansible/roles/metrics/files/prometheus.yml:14]: I would keep `devnet` for the production environment and maybe something like `pre-devnet` or `test-devnet` for the develop branch. 
We use `testnet` for the internal testnet

Comment (georgysavva) [reply deploy/ansible/roles/metrics/files/prometheus.yml:14]: I vote for `pre-devnet` then

File: .github/workflows/docker-develop.yml
Comment (georgysavva) [reply .github/workflows/docker-develop.yml:54]: As far as I know, we can add a review access policy for GitHub secrets:
https://docs.github.com/en/actions/reference/encrypted-secrets#reviewing-access-to-organization-level-secrets
It means that GHA cannot access the secret until you approve it.

Alternatively, we could create a user with the lowest privileges on host machines and allow GHA to connect via this user and not the root. This way the impact on host machines from GHA will be minimum 
@karimodm

Comment (georgysavva) [reply .github/workflows/docker-develop.yml:54]: @Wollac one more thing, here is a note from GHA docs:

> Note: With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository.
https://docs.github.com/en/actions/reference/encrypted-secrets#using-encrypted-secrets-in-a-workflow

So it means that only people who have **write** access to the Goshimmer repo can access secrets.

Comment (Wollac) [reply .github/workflows/docker-develop.yml:54]: > So it means that only people who have write access to the Goshimmer repo can access secrets.
But still, there are a lot more people with Repo access than people that should have access on the machine

File: .github/workflows/docker-develop.yml
Comment (georgysavva) [reply .github/workflows/docker-develop.yml:9]: It's just for debuggin, to run on this PR. I will remove it before merging

File: .github/workflows/docker-develop.yml
Comment (georgysavva) [reply .github/workflows/docker-develop.yml:54]: Then we should try GHA review access feature for secrets.

Comment (georgysavva) [reply .github/workflows/docker-develop.yml:54]: Ideally would be to restrict edit access to GHA workflow files, but it seems not possible for now. Here is an open issue requesting that:
https://github.com/actions/runner/issues/458

File: deploy/ansible/roles/metrics/files/prometheus.yml
Comment (luca-moser) [reply deploy/ansible/roles/metrics/files/prometheus.yml:14]: Lets go with `pre-devnet`

File: .github/workflows/docker-develop.yml
Comment (karimodm) [reply .github/workflows/docker-develop.yml:54]: In any case; I don't think this observation should be a blocker for the approval of this PR. We will have a separate discussion.

**Review by karimodm — CHANGES_REQUESTED** [review]
Nicely done @georgysavva!! First round of review, let's make it interactive!

File: deploy/ansible/roles/goshimmer-node/templates/docker-compose-analysis-server.yml.j2
Comment (georgysavva) [reply deploy/ansible/roles/goshimmer-node/templates/docker-compose-analysis-server.yml.j2:24]: I will do it the next PR when I generalize the deployment to accommodate the `devnet` environment as well.

File: deploy/ansible/roles/goshimmer-node/templates/docker-compose-analysis-server.yml.j2
Comment (georgysavva) [reply deploy/ansible/roles/goshimmer-node/templates/docker-compose-analysis-server.yml.j2:6]: Fixed

File: deploy/ansible/roles/goshimmer-node/templates/docker-compose-analysis-server.yml.j2
Comment (georgysavva) [reply deploy/ansible/roles/goshimmer-node/templates/docker-compose-analysis-server.yml.j2:1]: It's true for now. But it will change with the next PR when I will need to use variables for host URLs. So let's keep as template to not move it back and forth

File: deploy/ansible/roles/metrics/tasks/main.yml
Comment (georgysavva) [reply deploy/ansible/roles/metrics/tasks/main.yml:10]: Yes

File: deploy/ansible/roles/goshimmer-node/templates/docker-compose-goshimmer.yml.j2
Comment (georgysavva) [reply deploy/ansible/roles/goshimmer-node/templates/docker-compose-goshimmer.yml.j2:14]: The same as above:

> I will do it the next PR when I generalize the deployment to accommodate the devnet environment as well.

File: .github/workflows/docker-develop.yml
Comment (georgysavva) [reply .github/workflows/docker-develop.yml:65]: So we just need to delete the key file at the end of the step or there is simpler way?

File: deploy/ansible/roles/goshimmer-node/templates/docker-compose-entrynode.yml.j2
Comment (georgysavva) [reply deploy/ansible/roles/goshimmer-node/templates/docker-compose-entrynode.yml.j2:21]: You want a random network version to not mix with nodes from the previous deployment? 

The problem with using a random number is how do you know that number in the future when you want to connect to the network from the local machine e.g. @capossele is there a way to request the network version from a node by web API? 

BTW, we could use the SHA of the commit instead of a random number for the network version.

File: deploy/ansible/roles/metrics/files/docker-compose.yml
Comment (georgysavva) [reply deploy/ansible/roles/metrics/files/docker-compose.yml]: Done

File: deploy/ansible/roles/metrics/files/docker-compose.yml
Comment (georgysavva) [reply deploy/ansible/roles/metrics/files/docker-compose.yml]: Done

File: deploy/ansible/roles/metrics/files/prometheus.yml
Comment (georgysavva) [reply deploy/ansible/roles/metrics/files/prometheus.yml:5]: Done

File: .github/workflows/docker-develop.yml
Comment (karimodm) [reply .github/workflows/docker-develop.yml:65]: Yeah, maybe delete it as part of the `ansible-playbook` step.

**Review by karimodm — COMMENTED** [review]
Looks good so far: only thing is remote access to the services: you want to firewall all the published ports from outside the network of the hosts (`192.168.10.0/28`) except for Kibana; for which you have to make sure credentials must be provided to access it.
Fiddling with the Kibana configuration is very annoying: all this ELK stack security is a pretty mess.
You can always front Kibana with a Traefik that will take care of the HTTP basic auth for you with a middleware; I've sent you an example of this.

File: .github/workflows/docker-develop.yml
Comment (georgysavva) [reply .github/workflows/docker-develop.yml:65]: I wouldn't bother with it for now. Let's leave like that and improve the security part of workflows in the next iteration when we move them out of this repo.

**Review by karimodm — APPROVED** [review]

**Review by capossele — APPROVED** [review]

**Pull request merged**
